File: LC/lc905/lc905.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParity = function(nums) {
  let p1 = 0, p2 = (nums.length - 1), tmp;
  while (p1 < p2){
      if (nums[p1] % 2){
          tmp = nums[p1];
          nums[p1] = nums[p2];
          nums[p2] = tmp;
          p2 -= 1;
      } else {p1 += 1;}
  }
  return nums;
};

File: LC/lc905/lc905.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        l = 0
        r = len(A) - 1
        while l < r:
            if A[l] % 2:
                A[l], A[r] = A[r], A[l]
                r -= 1
            else:
                l += 1
        return A